arangod/Indexes/IndexIterator.h

```diff
@@ -69,8 +69,47 @@ class IndexIterator {
   friend class MultiIndexIterator;
 
  public:
+  class CoveringData {
+   public:
+    virtual VPackSlice at(size_t i) = 0;
```

Comment:
Not having a virtual destructor will work, as we don't seem to have any data members. But it may be better for the future to add a virtual destructor here, just in case this struct gets extended.